src/StaticAnalysis/ParsingFileAnalyser.php

```diff
@@ -71,12 +76,24 @@ final class ParsingFileAnalyser implements FileAnalyser
      */
     private $ignoreDeprecatedCode;
 
+    public static function computeHashForSource(string $source): int
+    {
+        return crc32($source);
```

Comment:
1. A single method to not duplicate code
2. PHP-CS-Fixer uses `crc32` without known issues: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/ae5701a7d22f3a808accf6e63e7b61abf8099ce6/src/Cache/FileCacheManager.php#L127